src/eyeTracker.m

```diff
@@ -55,161 +30,134 @@
                 el.displayCalResults = 1;
 
-                % call this function for changes to the calibration structure to take
-                % affect
+                % Call this function for changes to the calibration structure to take
+                %  affect.
                 EyelinkUpdateDefaults(el);
 
-                %% STEP 3
-                % Initialization of the connection with the Eyelink Gazetracker.
-                % exit program if this fails.
-
-                % Initialize EL and make sure it worked: returns: 0 if OK, -1 if error
-                ELinit = Eyelink('Initialize');
+                % Initialize EL and make sure it worked: returns 0 if OK, -1 if error.
+                %  Exit program if this fails.
+                elInit  = Eyelink('Initialize');
                 if ELinit ~= 0
                     fprintf('Eyelink is not initialized, aborted.\n');
                     Eyelink('Shutdown');
-                    Screen('CloseAll');
+                    CleanUp();
                     return
                 end
 
-                % Make sure EL is still connected: returns 1 if connected,
-                % -1 if dummy-connected,
-                % 2 if broadcast-connected and 0 if not connected
-                ELconnection = Eyelink('IsConnected');
+                % Make sure EL is still connected: returns 1 if connected, -1 if dummy-connected,
+                %  2 if broadcast-connected and 0 if not connected. Exit program if this fails.
+                elConnection = Eyelink('IsConnected');
                 if ELconnection ~= 1
                     fprintf('Eyelink is not connected, aborted.\n');
                     Eyelink('Shutdown');
-                    Screen('CloseAll');
+                    CleanUp();
                     return
                 end
 
-                % Last check that the EL is up to work
+                % Last check that the EL is up to work and exit program if this fails.
                 if ~EyelinkInit(0, 1)
                     fprintf('Eyelink Init aborted.\n');
+                    CleanUp();
                     return
                 end
 
-                % Open the edf file to write the data
+                % Open the edf file to write the data.
                 edfFile = 'demo.edf';
                 Eyelink('Openfile', edfFile);
 
+                % Get EyeLink setup information.
                 [el.v, el.vs] = Eyelink('GetTrackerVersion');
                 fprintf('Running experiment on a ''%s'' tracker.\n', el.vs);
 
-                % make sure that we get gaze data from the Eyelink
+                % Make sure that we get gaze data from the Eyelink.
                 Eyelink('Command', 'link_sample_data = LEFT,RIGHT,GAZE,AREA');
+                Eyelink('command', 'link_event_data = GAZE,GAZERES,HREF,AREA,VELOCITY');
+                Eyelink('command', 'link_event_filter = LEFT,RIGHT,FIXATION,BLINK,SACCADE,BUTTON');
```

Comment:
We need to dig into the help section to see if that means we can set which eye to follow.